home_audit.node.js

```diff
@@ -935,8 +935,9 @@ let validationRules = {
         return this._heating_efficiency_method(value, 2);
     },
     _heating_efficiency_method: function(value, num) {
-        if(value && (_homeValues['heating_type_'+num] === 'none' || TypeRules._is_empty(_homeValues['heating_type_'+num]))) {
-            return new Validation('Efficiency method should not be set if heating type is "none" or empty', ERROR);
+        const noEfficiencyMethod = ['baseboard', 'wood_stove', 'none'];
+        if(value && noEfficiencyMethod.indexOf(_homeValues['heating_type_'+num]) > -1 || (_homeValues['heating_type_'+num] === 'central_furnace' && _homeValues['heating_fuel_'+num] === 'electric') || TypeRules._is_empty(_homeValues['heating_type_'+num])) {
```

Comment:
This conditional is too long for my poor brain to understand. Could you break it into parts and assign them to variables to ease comprehension? Something like this:

```
const isHeatingTypeWithoutEfficiencyMethod = value && noEfficiencyMethod.indexOf(_homeValues['heating_type_'+num]) > -1;
const isElectricFurnace = _homeValues['heating_type_'+num] === 'central_furnace' && _homeValues['heating_fuel_'+num] === 'electric';
...
if (isHeatingTypeWithoutEfficiencyMethod || isElectricFurnace ...) {
    ...
}

Reply:
Separately: Shouldn't the check that `value` is truthy be independent of the heating type? Right now this conditional is true if e.g. the heater is an electric furnace, even if `value` is 0.

Reply:
You are correct.  And I suppose I shouldn't be doing truthy either, but instead using the TypeRules._is_empty() method